http_2.rflx

```diff
@@ -205,34 +204,39 @@ package HTTP_2 is
                if Frame_Type = HEADERS
                   and Flag_Padded = False and Payload_Length >= (Weight'Last - Exclusive_Flag'First + 1) / 8
             then null
-               if Frame_Type = PRIORITY;
+               if Frame_Type = PRIORITY and Message'Size = Payload_Length * 8 + (Stream_Identifier'Last + 1 - Payload_Length'First);
          Header_Block_Fragment : Opaque
             then Padding
                if (Frame_Type = HEADERS or Frame_Type = PUSH_PROMISE) and Flag_Padded = True
             then null
-               if Frame_Type = CONTINUATION
-                  or ((Frame_Type = PUSH_PROMISE or Frame_Type = HEADERS) and Flag_Padded = False);
+               if (Frame_Type = CONTINUATION
+                  or ((Frame_Type = PUSH_PROMISE or Frame_Type = HEADERS) and Flag_Padded = False))
+                  and Message'Size = Payload_Length * 8 + (Stream_Identifier'Last + 1 - Payload_Length'First);
          Application_Data : Opaque
             then Padding
                if Frame_Type = DATA and Flag_Padded = True
             then null
-               if (Frame_Type = DATA and Flag_Padded = False) or Frame_Type = PING;
+               if ((Frame_Type = DATA and Flag_Padded = False) or Frame_Type = PING)
+                  and Message'Size = Payload_Length * 8 + (Stream_Identifier'Last + 1 - Payload_Length'First);
          Padding : Opaque
             with Size => Pad_Length * 8;
          Error_Code : Error_Code
             then null
-               if Frame_Type = RST_STREAM
+               if Frame_Type = RST_STREAM and Message'Size = Payload_Length * 8 + (Stream_Identifier'Last + 1 - Payload_Length'First)
             then null
                if Frame_Type = GOAWAY
                   and Payload_Length * 8 = (Error_Code'Size + Last_Stream_ID'Size + Reserved_Bit_2'Size)
+                  and Message'Size = Payload_Length * 8 + (Stream_Identifier'Last + 1 - Payload_Length'First)
             then Additional_Debug_Data
                with Size => (Payload_Length * 8 - (Error_Code'Size + Last_Stream_ID'Size + Reserved_Bit_2'Size))
                if Frame_Type = GOAWAY
                   and Payload_Length * 8 > (Error_Code'Size + Last_Stream_ID'Size + Reserved_Bit_2'Size);
          Additional_Debug_Data : Opaque
-            then null;
+            then null
+               if Message'Size = Payload_Length * 8 + (Stream_Identifier'Last + 1 - Payload_Length'First);
```

Comment:
Each newly added condition should have a comment and a reference to Componolit/RecordFlux#554, so we can transform that into a global invariant later.

Reply:
Added in 65835c4